arrows/qt/widgets/MetadataView.cxx

```diff
@@ -0,0 +1,325 @@
+/*ckwg +29
+ * Copyright 2018-2020 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither the name Kitware, Inc. nor the names of any contributors may be
+ *    used to endorse or promote products derived from this software without
+ *    specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "MetadataView.h"
+
+#include <vital/types/metadata_traits.h>
+
+#include <vital/range/filter.h>
+#include <vital/range/valid.h>
+#include <vital/range/iota.h>
+
+#include <qtColorUtil.h>
+#include <qtSqueezedLabel.h>
+#include <qtStlUtil.h>
+
+#include <QEvent>
+#include <QHash>
+#include <QLabel>
+#include <QScrollBar>
+#include <QSet>
+#include <QSpacerItem>
+#include <QVariant>
+#include <QVBoxLayout>
+
+namespace kv = kwiver::vital;
+namespace kvr = kwiver::vital::range;
+
+namespace kwiver {
+namespace arrows {
+namespace qt {
+
+
+QTE_IMPLEMENT_D_FUNC(MetadataView)
+
+///////////////////////////////////////////////////////////////////////////////
+
+//BEGIN MetadataViewPrivate
+
+//-----------------------------------------------------------------------------
+class MetadataViewPrivate
+{
+public:
+  void addItem(int id, QString const& keyText);
+  void setItemValue(int id, QString const& valueText);
+  void clearItemValue(int id);
+
+  void removeItem(int id);
+  void clear();
+
+  QSet<int> itemIds() const { return this->keyLabels.keys().toSet(); }
+  void updateLabelColors();
+
+  QWidget* contentWidget;
```

Comment:
I realize this is a style change vs. TeleSculptor (whence this originated), but the KWIVER style is to add a trailing `_` to class members. Do we want to make that change?

Reply:
I don't think KWIVER is very consistent about that.  There are many places where members are prefixed with `m_` instead.  I'm okay with changing it if we want to be more consistent about member naming.  Then again, this whole file is full of other style violations.  I believe it is closer to Qt style than KWIVER style, and maybe that makes more sense for this subdirectory.  Thoughts?

Reply:
There are several styles for member variables. My personal preference is the `m_` prefix. There is some precedent for source files that are closely coupled with another package to use that style (e.g. VTK style), so I don't have a problem with this being more Qt like.

Reply:
My main preference would be to pick one and document it :slightly_smiling_face:. That said, it does appear that leading `m_` is more common than trailing `_`, and I also prefer it over trailing `_`. I wonder if the latter is largely a holdover from VITDK?

Note that Qt style would be `m_camelCase`.

> Then again, this whole file is full of other style violations.

Are there major violations other than inner-spaces and naming conventions? Other than naming conventions — which, yes, ought to follow Qt for Qt stuff, and this is done for the existing Qt arrow classes — it would be nice to fix these, though absent fully working integration with uncrustify, I'm usually willing to let spaces slide as long as a file is internally consistent.

Reply:
I was mostly referring to the fact that Qt style is not the same a KWIVER style, primarily regarding the use of camelCase vs underscores.  Is `m_camelCase` something that matches Qt style?  It seems like a mixing of styles, but if that's what is done in Qt and that's also what we do in KWIVER, then we should do that here too.

Reply:
>  Is `m_camelCase` something that matches Qt style?

That's... actually a fair question. There is no doubt it has been used historically (note: I'm talking about *inside Qt itself*, not just other projects *using* Qt), but it also appears that they may be moving away from it in favor of undecorated (but still camelCase) member names. In any case, I have no problem with the convention and am using it in at least one new project.

Reply:
I've addressed this now, switched to the `m_` prefix.  I think all other issues are addressed as well.